modules/ardrivo/test/Arduino.cxx

```diff
@@ -0,0 +1,311 @@
+          "[isDigit]") {
+    REQUIRE(isDigit('1'));
+    REQUIRE(!isDigit('a'));
+    REQUIRE(!isDigit('#'));
+}
+
+TEST_CASE("Checks if a character is a graphical represention"
+          "[isGraph]") {
+    REQUIRE(isGraph('a'));
+    REQUIRE(isGraph('1'));
+    REQUIRE(!isGraph(' '));
+}
+
+TEST_CASE("Checks if a character is hexadecimal", "[isHexadecimalDigit]") {
+    REQUIRE(isHexadecimalDigit('a'));
+    REQUIRE(!isHexadecimalDigit('!'));
+}
+
+TEST_CASE("Checks if a character is lowercase", "[isLowerCase") {
+    REQUIRE(isLowerCase('a'));
+    REQUIRE(!isLowerCase('A'));
+}
+
+TEST_CASE("Checks if a character is printable", "[isPrintable]") {
+    REQUIRE(isPrintable('a'));
+    REQUIRE(isPrintable('1'));
+    REQUIRE(isPrintable(' '));
+}
+
+TEST_CASE("Checks if a character is a punctuation"
+          "[isPunct") {
+    REQUIRE(!isPunct('a'));
+    REQUIRE(isPunct(','));
+    REQUIRE(isPunct('?'));
+}
+
+TEST_CASE("Checks if a character contains white-space", "[isSpace]") {
+    REQUIRE(!isSpace('a'));
+    REQUIRE(isSpace('\t'));
+    REQUIRE(isSpace(' '));
+}
+
+TEST_CASE("Checks if a character is uppercase"
+          "[isUpperCase") {
+    REQUIRE(isUpperCase('A'));
+    REQUIRE(!isUpperCase('a'));
+}
+
+TEST_CASE("Checks if a character is a blank character"
+          "[isWhiteSpace]") {
+    REQUIRE(isWhitespace(' '));
+    REQUIRE(isWhitespace('\t'));
+    REQUIRE(!isWhitespace('a'));
+}
+
+TEST_CASE("Checks if it returns a random number within a given range"
+          "[isRandom]") {
+    auto i = GENERATE(take(100, filter([](int i) { return i; }, random(-100, 100))));
+    REQUIRE(i < 100);
+    REQUIRE(i > -100);
+}
+
+TEST_CASE("Test randomSeed", "[randomSeed]") {
+    std::array<long, 100> arr{};
+    std::array<long, 100> arr2{};
+    std::array<long, 100> arr3{};
+
+    auto count = 0;
+    auto iterator = 0;
+    randomSeed(2);
+
+    std::generate(arr.begin(), arr.end(), []() { return random(100); });
+    randomSeed(3);
+    std::generate(arr2.begin(), arr2.end(), []() { return random(100); });
+
+    for (int i = 0; i < 100; i++) {
+        if (arr == arr2) {
+            count++;
+        }
+    }
+    REQUIRE(count < 3);
+
+    randomSeed(2);
+
+    std::generate(arr3.begin(), arr3.end(), []() { return random(100); });
+
+    for (int i = 0; i < 100; i++) {
+        REQUIRE(arr == arr3);
+    }
+}
+
+void fun() {}
+
+TEST_CASE("Check if vector contains interupt", "[attachInterrupt]") {
+    // First test
+    auto (*user_func)() = &fun;
+    attachInterrupt(0, (user_func), 1);
+    auto test = board_data->interrupts_handlers[0];
+    REQUIRE(test.first != NULL);
+    REQUIRE(test.second == 1);
+
+    // Second test
+    attachInterrupt(2, (*user_func), 2);
+    auto test2 = board_data->interrupts_handlers[2];
+    REQUIRE(test2.first != NULL);
+    REQUIRE(test2.second == 2);
+}
+
+TEST_CASE("Delete a interupt from vector", "[detachInterrupt]") {
+    auto test = board_data->interrupts_handlers[0];
```

Comment:
Use a structured binding e.g.
```cpp
auto [func, pin] = board_data->interrupts_handlers[0];
```
this will split the std::pair into 2 variables so that you dont have to use .first and .second